syntax = "proto3";

package account;

message Account {
  string id = 1;
  string username = 2;
  string password = 3;
  int32 highScore = 4;
  string account_type = 5;
}

message AccountRequest {
  string username = 1;
}

message Empty {}

service AccountService {
  rpc CreateAccount (Account) returns (Empty);
  rpc GetAccount (AccountRequest) returns (Account);
  rpc UpdateAccount (Account) returns (Empty);
  rpc DeleteAccount (Account) returns (Empty);
  rpc UpdateHighScore (Account) returns (Empty);
  rpc UpdateAccountType (Account) returns (Empty);
}
